lib/dams/share_api.rb

```diff
@@ -0,0 +1,203 @@
+          document.tags = osf_mads_fields(@document)
+          osf_contributors(@document).each do |contributor|
+            document.add_contributor(contributor)
+          end
+          
+          if document.valid?
+            @headers = {'Authorization' => "Token #{share_config.fetch('token')}", 
+                       'Content-Type'  => 'application/json'
+                      }
+
+            @route = "#{share_config.fetch('host')}api/v1/share/data"
+            @response = with_timeout { HTTParty.post(@route, body: document.to_share.to_json, headers: @headers)}
+            msg = "Success"
+          else
+            msg = "Failed"
+          end
+        end
+      rescue Exception => e
+        p e
+      end
+    end
+
+ 
+    def self.share_config
+      env = Rails.env || 'test'
+      @config ||= YAML.load(ERB.new(IO.read(File.join(Rails.root, 'config', 'share_notify.yml'))).result)[env].with_indifferent_access
+    end
+
+    def self.with_timeout(&_block)
+      Timeout.timeout(5) { yield }
+    end
+
+    #Mapping for OSF API
+    def self.osf_title(document)
+        field_name = "title_json_tesim"
+        dams_data= document["#{field_name}"]
+        osf_data=''
+
+      if dams_data != nil
+        dams_data.each do |datum|
+          title = JSON.parse(datum)
+          osf_data = title['name'] ? title['name'] : ''
+          osf_data += title['name'] && !title['translationVariant'].blank? ? ' : ' : ''
+          title_trans = title['translationVariant'] || []
+          if title_trans.class == Array
+            title_trans.each do |trans|  
+              osf_data += trans
+            end
+          elsif title_trans.class == String
+            osf_data += title_trans
+          end
+        end
+      end
+      osf_data
+    end
+
+    def self.osf_contributors(document)
+      field_name = "relationship_json_tesim"
+      dams_data = document["#{field_name}"]
+      osf_data =[]
+
+      if dams_data != nil
+        dams_data.each do |datum|
+          
+          relationships = JSON.parse(datum)
+          relationships.each do |key, value|
+            value.each do |v|
+              osf_data << {"name": v}
+            end
+          end
+        end
+      end
+      osf_data = (osf_data.blank?) ? osf_data << {"name": "UC San Diego Library"} : osf_data
+    end
+
+    def self.osf_description(document)
+      field_name = "otherNote_json_tesim"
+      dams_data = document["#{field_name}"]
+      osf_data = ''
+
+      if dams_data != nil
+        dams_data.each do |datum|
+          other_note = JSON.parse(datum)
+          osf_data = other_note['value'] if other_note['type'] == 'description'
+        end
+      end
+      osf_data
+    end 
+
+    def self.osf_uris(document)
+      field_name = "id"
+      dams_data = document["#{field_name}"]
+      osf_data = {}
+
+      if dams_data != nil
+        url = "http://library.ucsd.edu/dc/collection/#{dams_data}"
```

Comment:
@hweng - you already could access the field Rails.configuration.host_name , I'm wondering if it could be used here instead of the hardcode url. Same question for the line 186 
 +      osf_data = {"name": "UC San Diego Library, Digital Collections", "uri": "http://library.ucsd.edu/dc"}